### PR TITLE
refactor(avoidance): generate drivable lanes in utils

### PR DESCRIPTION
**Pull request by satoshi-ota**
### <samp>🤖 Generated by Copilot at b55577d</samp>

Refactored some avoidance-related functions in the behavior path planner. Moved some helper functions to `utils.cpp` and renamed `generateExtendedDrivableArea` to `generateExpandDrivableLanes` to improve code reuse and readability. Added the new function declaration to `utils.hpp`.

## Tests performed

- [x] [PASS TIER IV INTERNAL SCENARIOS no.1](https://evaluation.tier4.jp/evaluation/reports/effbf2fe-b763-5867-8ea8-a6264dfe47f8?project_id=prd_jt)
- [x] [PASS TIER IV INTERNAL SCENARIOS no.2](https://evaluation.tier4.jp/evaluation/reports/0cd7d984-1e68-5665-9991-46eed8dd0fb6?project_id=prd_jt)

## Effects on system behavior

Nothing.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/src/utils/avoidance/utils.cpp [hunk 1]
```diff
@@ -1677,4 +1693,145 @@ std::pair<PredictedObjects, PredictedObjects> separateObjectsByPath(
...
+
+  if (!parameters->use_adjacent_lane) {
+    return current_drivable_lanes;
+  }
+
+  const auto has_same_lane =
+    [](const lanelet::ConstLanelets lanes, const lanelet::ConstLanelet & lane) {
+      if (lanes.empty()) return false;
+      const auto has_same = [&](const auto & ll) { return ll.id() == lane.id(); };
+      return std::find_if(lanes.begin(), lanes.end(), has_same) != lanes.end();
```
Comment (zulfaqar-azmi-t4) [inline planning/behavior_path_planner/src/utils/avoidance/utils.cpp:1715]: using `std::any_of` would give more accurate nuance.

Comment (satoshi-ota) [reply planning/behavior_path_planner/src/utils/avoidance/utils.cpp]: @zulfaqar-azmi-t4 Thanks for your advice! I fixed in https://github.com/autowarefoundation/autoware.universe/pull/5018/commits/e9fe60525d3e6510ba7957bf91b53280da726f47.

Comment (zulfaqar-azmi-t4) [reply planning/behavior_path_planner/src/utils/avoidance/utils.cpp]: thank you Ota-san!

**Review by zulfaqar-azmi-t4 — APPROVED** [review]
LGTM!

**Pull request merged**